### ISSUE TITLE
Type stability

### ISSUE THREAD

**Issue by brendanjohnharris** [brendanjohnharris/TimeseriesFeatures.jl#10]
How specific can we be with Feature input and return types? How specific should we be?
Currently:
```
μ = Feature(mean, :mean, ["distribution"], "μ")
x = randn(1000) .|> Float32
@inferred getmethod(μ)(x) # Ok
@inferred μ(x) # Not ok. The inferred type is Any, the return type is Float32. Can---should---we fix this
```

**Comment by brendanjohnharris** [comment]
See https://discourse.julialang.org/t/julia-vs-python-performance-on-package-catch22-why-is-julia-slower-in-this-case-and-how-can-i-improve-it

**Comment by brendanjohnharris** [comment]
Resolved in v0.6.0

**Comment by brendanjohnharris** [comment]
Still not type stable when indexing feature sets
